kernel/vmemitcodes.ml

```diff
@@ -300,14 +300,23 @@ let check_prim_op = function
   | Float64ldshiftexp -> opCHECKLDSHIFTEXP
   | Float64next_up    -> opCHECKNEXTUPFLOAT
   | Float64next_down  -> opCHECKNEXTDOWNFLOAT
-  | Arraymake | Arrayget | Arrayset | Arraydefault | Arraycopy | Arraylength ->
-    assert false
+  | Arraymake | Arrayget | Arrayset | Arraydefault | Arraycopy | Arraylength
+  | Charcode | Charchr
+  | Stringmake | Stringlength | Stringget | Stringsub | Stringcat
+  | Stringequal | Stringlt | Stringle
+    -> assert false
 
 let check_caml_prim_op = function
 | CAML_Arraymake -> opCHECKCAMLCALL2_1
 | CAML_Arrayget -> opCHECKCAMLCALL2
 | CAML_Arrayset -> opCHECKCAMLCALL3_1
 | CAML_Arraydefault | CAML_Arraycopy | CAML_Arraylength -> opCHECKCAMLCALL1
+| CAML_Charcode | CAML_Charchr -> opCHECKCAMLCALL1
+| CAML_Stringmake -> opCHECKCAMLCALL2_1
```

Comment:
No, the last argument cannot be an accumulator. I don't think it would trigger a segfault, but it would at least make it trivial to prove false. So, you have to use `opCHECKCAMLCALL2` here.